src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -688,7 +688,12 @@ void Lowering::ContainBlockStoreAddress(GenTreeBlk* blkNode, unsigned size, GenT
     {
         return;
     }
-#endif // TARGET_ARM
+#else // !TARGET_ARM
+    if ((ClrSafeInt<int>(offset) + ClrSafeInt<int>(size)).IsOverflow())
```

Comment:
Can this overflow on arm too? Should this be outside the ifdef?

Reply:
I think the ARM specific check above is fine -- we only get here for unrolled block copies so `size` is guaranteed to be small.